### DISCUSSION TITLE
Migrating from Orleans 3.x to 7.0

### DISCUSSION THREAD

**Discussion by ramrafi** [dotnet/orleans#8425] [Q&A] [answered]
Hi,

After going through the official migration guide on https://learn.microsoft.com/en-us/dotnet/orleans/migration-guide and successfully connecting client and server we are dealing with an annoying problem.

The problem is that we cannot resolve any grain from the client class, every attempt ends with the exception: 'System.InvalidOperationException: Unable to find an IGrainReferenceActivatorProvider for grain type <our_grain_type>' that comes from IGrainFactory/IClusterClient.GetGrain<> method.

Any ideas on how to resolve this would be much appreciated !

**Comment by ReubenBond** [comment, marked as answer]
This should be an issue. I think it's likely that the problem you're facing is that `Microsoft.Orleans.Sdk` is not installed in all of your projects (all projects related to Orleans, including client hosts, server hosts, grain interfaces, grain implementations, serializable types). Could you double-check that?

**Reply by ramrafi** [reply]
Thank you Reuben for your input, we found the problem!

**Reply by perfectphase** [reply]
This just caught me too, just had the Abstractions project in with my interfaces.

**Reply by derodriguezat** [reply]
Thank you Reuben! I had the same issue because I did not to include Microsoft.Orleans.Sdk in the interfaces project.

**Comment by ElanHasson** [comment]
I've seen this topic come up a few times, perhaps this is a candidate for an analyzer?
